### DISCUSSION TITLE
CT enhancement

### DISCUSSION THREAD

**Discussion by lxsxl** [PE5PVB/TEF6686_ESP32#345] [Ideas]
As there is a discussion on discord, CT function could possibly be adjusted like this:

1. Once CT is received keep it running on the TEF even if tuning to a different frequency is done (I think it was already like this for a short time)
2. If tuned to a different frequency remove CT-indicator and switch time-colour from yellow to blue (so it will be visible for the "raw-data-lovers" that the time is not from the actually tuned frequency).
3. If the recently tuned frequency transmits group 4A/CT then re-indicate CT and re-switch time-colour to yellow (so it will be visible again that CT it is from the actual frequency).

**Comment by lxsxl** [comment]
Just saw that the CT-indicator was removed already. So anything to this can be skipped.

**Comment by PE5PVB** [comment]
Currently the clock is shown as transmitted. When no CT data, no clock is shown. It's going to be a lot of if-then-else situations and it can confuse the user more. If you want to know the time, just watch your phone?

**Reply by lxsxl** [reply]
For me it is okay like it is.
I only noticed discussion on discord and thought it might be a good idea to keep the time running. So let's see if it will remain in discussion or if it will take care of itself.

**Comment by PE5PVB** [comment]
Just added: Once when CT has been received, time is synced to internal RTC. When CT is not avaialble the RTC time will be shown in secondary color. This clock is also visible in AM mode.

**Comment by lxsxl** [comment]
Looks fine. But the time is also displayed in the bottom of the EON PI column. I think this was not intended?

**Comment by lxsxl** [comment]
Another thing: If CT data were received and then RDS get‘s lost the time freezes. If then tuned to a different frequency there will be the meantime digits in overlay. (If CT was received without loss of RDS and then tuned to a different frequency the time keeps running.)

**Comment by PE5PVB** [comment]
Correct. On RDS loss the time is frozen, even like with PS, PTY RT etcetera. But it should not be overwritten faulty, I will look at this tomorrow.
